src/librustc_typeck/astconv.rs

```diff
@@ -385,8 +385,15 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
             let (spans, label) = if required == permitted && provided > permitted {
                 // In the case when the user has provided too many arguments,
                 // we want to point to the unexpected arguments.
-                let spans: Vec<Span> = args.args[offset + permitted..offset + provided]
+                let spans: Vec<Span> = args.args
                     .iter()
+                    .filter(|arg| match (kind, arg) {
```

Comment:
IMO it might be better to use `GenericParamOrd` or something instead of strings for `kind` here

Reply:
Er should that go in this PR?

Reply:
It might be better to do so if we don't want to forget this